icu4j/main/core/src/main/java/com/ibm/icu/segmenter/LocalizedSegmenter.java

```diff
@@ -0,0 +1,188 @@
+  }
+
+  /**
+   * @return a builder for constructing {@code LocalizedSegmenter}
+   * @draft ICU 78
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  LocalizedSegmenter(ULocale locale, SegmentationType segmentationType) {
+    this.locale = locale;
+    this.segmentationType = segmentationType;
+  }
+
+  /**
+   * @internal
+   * @deprecated This API is ICU internal only.
+   */
+  @Override
+  @Deprecated
+  public BreakIterator getNewBreakIterator() {
+    BreakIterator breakIter;
+    switch (this.segmentationType) {
+      case LINE:
+        breakIter = BreakIterator.getLineInstance(this.locale);
+        break;
+      case SENTENCE:
+        breakIter = BreakIterator.getSentenceInstance(this.locale);
+        break;
+      case WORD:
+        breakIter = BreakIterator.getWordInstance(this.locale);
+        break;
+      case GRAPHEME_CLUSTER:
+      default:
+        breakIter = BreakIterator.getCharacterInstance(this.locale);
+        break;
+    }
+    return breakIter;
+  }
+
+  /**
+   * The type of segmentation to be performed. See the ICU User Guide page
+   * <a
+   * href="https://unicode-org.github.io/icu/userguide/boundaryanalysis/#four-types-of-breakiterator">Boundary Analysis</a>
+   * for further details.
+   */
+  public enum SegmentationType {
+    GRAPHEME_CLUSTER,
+    WORD,
+    LINE,
+    SENTENCE,
+  }
+
+  /**
+   * Builder for {@link LocalizedSegmenter}
+   * @draft ICU 78
+   */
+  public static class Builder {
+
+    private ULocale locale = ULocale.ROOT;
+
+    private SegmentationType segmentationType = SegmentationType.GRAPHEME_CLUSTER;
+
+    Builder() { }
+
+    /**
+     * Set the locale for which segmentation rules will be loaded
+     * @param locale an ICU locale object
+     * @draft ICU 78
+     */
+    public Builder setLocale(ULocale locale) {
+      this.locale = locale;
+      return this;
+    }
+
+    /**
+     * Set the locale for which segmentation rules will be loaded
+     * @param locale a Java locale object
+     * @draft ICU 78
+     */
+    public Builder setLocale(Locale locale) {
+      this.locale = ULocale.forLocale(locale);
+      return this;
+    }
+
+    /**
+     * Set the segmentation type to be performed.
+     * @param segmentationType
+     * @draft ICU 78
+     */
+    public Builder setSegmentationType(SegmentationType segmentationType) {
+      this.segmentationType = segmentationType;
+      return this;
+    }
+
+    /**
+     * Builds the {@code Segmenter}
+     * @return the constructed {@code Segmenter} instance
+     * @draft ICU 78
+     */
+    public Segmenter build() {
+      return new LocalizedSegmenter(this.locale, this.segmentationType);
+    }
+
+  }
+
+  class LocalizedSegments implements Segments {
+
+    private CharSequence source;
+
+    private LocalizedSegmenter segmenter;
+
+    private BreakIterator breakIter;
+
+    private LocalizedSegments(CharSequence source, LocalizedSegmenter segmenter) {
```

Comment:
nit/optional: i would swap the parameters

Reply:
Done.